lib/collection.js

```diff
@@ -737,23 +769,37 @@ Collection.prototype.insert = deprecate(function(docs, options, callback) {
  * @param {Collection~updateWriteOpCallback} [callback] The command result callback
  * @return {Promise} returns Promise if no callback passed
  */
+const updateOneSchema = {
+  upsert: { type: 'boolean' },
+  w: { type: ['number', 'string'] },
+  wtimeout: { type: 'number' },
+  j: { type: 'boolean' },
+  bypassDocumentValidation: { type: 'boolean' },
+  arrayFilters: { type: 'array' },
+  collation: { type: 'object' },
+  ignoreUndefined: { type: 'boolean', overrideOnly: true },
+  session: { type: ClientSession },
+  multi: { overrideOnly: true }
+};
 Collection.prototype.updateOne = function(filter, update, options, callback) {
   if (typeof options === 'function') (callback = options), (options = {});
-  options = options || {};
 
   const err = checkForAtomicOperators(update);
   if (err) {
     if (typeof callback === 'function') return callback(err);
     return this.s.promiseLibrary.reject(err);
   }
 
-  options = Object.assign({}, options);
-
-  // Add ignoreUndfined
-  if (this.s.options.ignoreUndefined) {
-    options = Object.assign({}, options);
-    options.ignoreUndefined = this.s.options.ignoreUndefined;
-  }
+  options = validate(
+    updateOneSchema,
+    options,
+    {},
+    {
+      ignoreUndefined: this.s.options.ignoreUndefined,
```

Comment:
In `validate`, we should be checking that overrides/dynamic defaults being applied have values (i.e. `!= null`).